commons/src/main/java/com/powsybl/commons/datasource/TarDataSource.java

```diff
@@ -0,0 +1,316 @@
+                 BufferedInputStream bis = new BufferedInputStream(fis);
+                 InputStream is = getCompressedInputStream(bis, compressionFormat);
+                 TarArchiveInputStream tais = new TarArchiveInputStream(is)) {
+
+                TarArchiveEntry entry;
+                while ((entry = tais.getNextEntry()) != null) {
+                    if (entry.getName().equals(fileName)) {
+                        return true;
+                    }
+                }
+                return false;
+            } catch (IOException | UnsupportedOperationException e) {
+                return false;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Check if a file exists in the archive. The file name will be constructed as:
+     * <p>{@code <basename><suffix>.<ext>}</p>
+     * @param suffix Suffix to add to the basename of the datasource
+     * @param ext Extension of the file (for example: .iidm, .xml, .txt, etc.)
+     * @param checkConsistencyWithDataSource Should the filename be checked for consistency with the DataSource
+     * @return true if the file exists, else false
+     */
+    @Override
+    public boolean exists(String suffix, String ext, boolean checkConsistencyWithDataSource) throws IOException {
+        return exists(DataSourceUtil.getFileName(baseName, suffix, ext), checkConsistencyWithDataSource);
+    }
+
+    /**
+     * Check if a file exists in the archive.
+     * @param fileName Name of the file
+     * @param checkConsistencyWithDataSource Should the filename be checked for consistency with the DataSource
+     * @return true if the file exists, else false
+     */
+    @Override
+    public boolean exists(String fileName, boolean checkConsistencyWithDataSource) {
+        Objects.requireNonNull(fileName);
+        Path tarFilePath = getArchiveFilePath();
+        return (!checkConsistencyWithDataSource || isConsistentWithDataSource(fileName)) && entryExists(tarFilePath, fileName);
+    }
+
+    @Override
+    public OutputStream newOutputStream(String suffix, String ext, boolean append) throws IOException {
+        return newOutputStream(DataSourceUtil.getFileName(baseName, suffix, ext), append);
+    }
+
+    @Override
+    public OutputStream newOutputStream(String fileName, boolean append) throws IOException {
+        Objects.requireNonNull(fileName);
+        if (append) {
+            throw new UnsupportedOperationException("append not supported in tar file data source");
+        }
+        Path tarFilePath = getArchiveFilePath();
+        OutputStream os = new TarEntryOutputStream(tarFilePath, fileName, compressionFormat);
+        return observer != null ? new ObservableOutputStream(os, tarFilePath + ":" + fileName, observer) : os;
+    }
+
+    @Override
+    public InputStream newInputStream(String suffix, String ext, boolean checkConsistencyWithDataSource) throws IOException {
+        return newInputStream(DataSourceUtil.getFileName(baseName, suffix, ext));
```

Comment:
should pass checkConsistencyWithDataSource ?

Reply:
fixed